### PR TITLE
fix(core/xref): use empty data-cite for local references

### PR DESCRIPTION
**Pull request by sidvishnoi**

### CODE REVIEW

File: tests/spec/core/xref-spec.js [hunk 1]
```diff
@@ -65,14 +65,24 @@ describe("Core — xref", () => {
         <p>As <a data-cite="!infra">ASCII uppercase</a> is valid dfn, it resolves to fragment. a local data-cite (infra) overrides parent's datacite.</p>
 
         <p>As <a data-cite="!infra">ASCII uppercasing</a> doesn't exist in INFRA, it resolves to spec only.</p>
+
+        <!-- local links with empty parent data-cite -->
```
Comment (marcoscaceres) [inline tests/spec/core/xref-spec.js]: This test might be getting a little big... would be good to split it, but have both conditions tested: that is, local link, and external link inside the section.

File: src/core/link-to-dfn.js [hunk 2]
```diff
@@ -125,10 +127,17 @@ export async function run(conf, doc, cb) {
     }
   });
 
+  const additionaExternalLinks = [
```
Comment (marcoscaceres) [inline src/core/link-to-dfn.js]: A comment here would be nice, saying what "additionExternalLinks" are.

File: src/core/link-to-dfn.js [hunk 3]
```diff
@@ -125,10 +127,17 @@ export async function run(conf, doc, cb) {
...
+    ...document.querySelectorAll(
+      "a[data-cite]:not([data-cite='']):not([data-cite*='#'])"
+    ),
+  ].filter(el => {
```
Comment (marcoscaceres) [inline src/core/link-to-dfn.js]: Looks like this is mapping? 

.map(el=> el.closest("[data-cite]")).filter(closest = > !closest || closest.dataset.cite !== "")

Comment (sidvishnoi) [reply src/core/link-to-dfn.js]: No. Not a mapping here.
The present one takes into account links of type `<section data-cite="foo"><a>bar</a></section>` and `<section><a data-cite="foo">bar</a></section>`. 
What we need in `additionalLinks` is the `el` itself, and not `el.closest()`

Comment (marcoscaceres) [reply src/core/link-to-dfn.js]: ok, cool.

**Pull request merged**